### PR TITLE
Upgrade shared framework to 1.0.0 RTM

### PR DESCRIPTION
**Pull request by dagood**
Upgrades dependencies and runtimeconfig.json files to use the RTM shared framework. I was able to build corefx using these changes on windows, and an OSX build I have going now looks fine. (After upgrading CLI).

@joperezr In https://github.com/dotnet/buildtools/pull/931#discussion_r73383924 you said we need to:
> - Make all of the tools compile against RTM shared framework since that is what they will have at runtime. (This is the hard part, since it is not only updating the dependencies in buildtools, but also the ones for msbuild and for roslyn)

It appears that msbuild at least works fine with a Microsoft.NETCore.App 1.0.0 runtimeconfig.json--are there specific scenarios that I need to test? Maybe I don't understand what you mean by updating the dependencies for msbuild and roslyn.

This PR also adds the new dependency update system because it came in handy when updating the project.jsons.

Addresses https://github.com/dotnet/buildtools/issues/954.

Related further work to hopefully unpin the version entirely: https://github.com/dotnet/buildtools/issues/918.

After getting a package with these changes, we should upgrade the version of CLI buildtools uses and take the new package so that work isn't left to whoever needs to consume a buildtools change in buildtools later on.

/cc @weshaggard @crummel @ericeil @jhendrixMSFT 

---

Tested | Win | OSX
--- | --- | ---
corefx | :ok: | :ok:
coreclr | :ok: | :ok:
wcf | :ok: | :ok:
core-setup | :ok: | :ok:

:ok: = running default build script works without errors

### CODE REVIEW

**Comment by joperezr** [comment]
> Maybe I don't understand what you mean by updating the dependencies for msbuild and roslyn.

What I meant is that just as you updated the dependencies of all of the tools in buildtools with this PR, we should probably do the same with msbuild and roslyn. The reason is that by this you ensure that at least you compiled against the same version of the assembly that you will have at runtime. This is not super necessary as you have seen since you were able to build corefx with this, but we might have added a breaking change from rc3 version of the shared library to RTM which is why I suggested to update MSBuild and Roslyn dependencies as well. Anyway, if corefx and coreclr with these changes, then I'm fine with not updating msbuild and the compilers.

**Comment by weshaggard** [comment]
For now lets decouple the update to msbuild and roslyn from this update. I think @chcosta has already updated to a post-RTM version of roslyn anyway.

**Comment by weshaggard** [comment]
The changes LGTM (assuming we have got all the places and they work).

**Comment by dagood** [comment]
@joperezr I see, thanks, I was thinking that might have been an all-or-nothing situation.

I've added a testing table to the description. (I mentioned earlier offline that I thought corefx on OSX didn't work, but that turned out to my firewall hanging the network tests.)

What I hit in coreclr looks like it might be incompatibility with https://github.com/dotnet/buildtools/pull/979 (cc @ericstj). [An excerpt from diagnostic logs, before/after](https://gist.github.com/dagood/2235441f6d4d011185d374b17ffe7038). Is there something I'd need to change to integrate this into coreclr? (If that's really the problem?) I tried buildtools `1.0.26-prerelease-00802-01` with no CLI upgrade and it also had this problem.

**Comment by ericstj** [comment]
That's odd, looks like CoreCLR isn't passing in the StablePackages list or the PackageIndex.  CoreCLR does do this in rel: https://github.com/dotnet/coreclr/blob/release/1.0.0/src/.nuget/stable.packages.props.  Simple workaround would be to add this item.  I will also fix the tasks to be tolerant of no StablePackage list.

**Comment by dagood** [comment]
After adding that file and the dir.props to direct to it, I get:
```
Task "ApplyBaseLine" (TaskId:18)
  Task Parameter:Apply=True (TaskId:18)
C:\git\coreclr\Tools\Packaging.targets(717,5): error MSB4018: The "ApplyBaseLine" task failed unexpectedly. [C:\git\coreclr\src\.nuget\Microsoft.NETCore.Jit\win\Microsoft.NETCore.Jit.pkgproj]
C:\git\coreclr\Tools\Packaging.targets(717,5): error MSB4018: System.NullReferenceException: Object reference not set to an instance of an object. [C:\git\coreclr\src\.nuget\Microsoft.NETCore.Jit\win\Microsoft.NETCore.Jit.pkgproj]
C:\git\coreclr\Tools\Packaging.targets(717,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.ApplyBaseLine.GetBaseLinedDependenciesFromBaseLinePackages() [C:\git\coreclr\src\.nuget\Microsoft.NETCore.Jit\win\Microsoft.NETCore.Jit.pkgproj]
C:\git\coreclr\Tools\Packaging.targets(717,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.ApplyBaseLine.Execute() [C:\git\coreclr\src\.nuget\Microsoft.NETCore.Jit\win\Microsoft.NETCore.Jit.pkgproj]
C:\git\coreclr\Tools\Packaging.targets(717,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [C:\git\coreclr\src\.nuget\Microsoft.NETCore.Jit\win\Microsoft.NETCore.Jit.pkgproj]
C:\git\coreclr\Tools\Packaging.targets(717,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext() [C:\git\coreclr\src\.nuget\Microsoft.NETCore.Jit\win\Microsoft.NETCore.Jit.pkgproj]
Done executing task "ApplyBaseLine" -- FAILED. (TaskId:18)
```
Looks similar, none of the items are given, but not sure what the expectation is.

**Comment by dagood** [comment]
@MattGal I've uploaded my build of this change to https://www.myget.org/feed/dagood-test-buildtools/package/nuget/Microsoft.DotNet.BuildTools/1.0.26-prerelease-00801-0, do you mind checking to make sure it works with Helix?

**Comment by dagood** [comment]
@ericstj I ran into one more, added a fix with https://github.com/dotnet/buildtools/pull/1005/commits/28dbe2a9dfa56456ea98fdbc962083d71aa2101b.

**Comment by MattGal** [comment]
Change LGTM from Helix perspective;  no one's ever broken Helix execution just changing what you've changed.

**Comment by dagood** [comment]
The build run table is all :ok:!

This is more of a FYI because it happens using old buildtools too, but while I was upgrading core-setup, I saw a binding error when using Desktop msbuild. For [pack.cmd](https://github.com/dotnet/core-setup/blob/b6c4ba424488c0e710e0ae0675a55edf68041554/pkg/pack.cmd) I first changed from `corehost msbuild.exe` to `msbuild.cmd`, and got these binding errors: https://gist.github.com/dagood/27ca2a00ff3f5ceb4da8a1b331c1a702. `dotnet.exe msbuild.exe` works (and in this case is the more natural upgrade change) but I'm wondering if we intend for this to work with Desktop msbuild.

@ericstj Could you review the PR's version changes too?

File: src/Microsoft.DotNet.Build.Tasks.Packaging/src/ApplyPreReleaseSuffix.cs [hunk 1]
```diff
@@ -62,7 +62,7 @@ public override bool Execute()
                 return false;
             }
 
-            if (PackageIndexes != null || PackageIndexes.Length > 0)
```
Comment (ericstj) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/src/ApplyPreReleaseSuffix.cs]: Can you instead change this to &&

**Comment by ericstj** [comment]
Version changes LGTM

File: src/Microsoft.DotNet.Build.Tasks.Packaging/src/ApplyPreReleaseSuffix.cs (hunk 1 above)
Comment (dagood) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/src/ApplyPreReleaseSuffix.cs]: Sure.

**Comment by dagood** [comment]
Thanks all! I'm going to submit upgrade PRs to the repos I listed in the description.

**Pull request merged**